src/library_tty.js

```diff
@@ -100,15 +100,26 @@ mergeInto(LibraryManager.library, {
             var buf = new Buffer(BUFSIZE);
             var bytesRead = 0;
 
+            var isPosixPlatform = (process.platform != 'win32'); // Node doesn't offer a direct check, so test by exclusion
+
             var fd = process.stdin.fd;
-            // Linux and Mac cannot use process.stdin.fd (which isn't set up as sync)
-            var usingDevice = false;
-            try {
-              fd = fs.openSync('/dev/stdin', 'r');
-              usingDevice = true;
-            } catch (e) {}
+            if (isPosixPlatform) {
+              // Linux and Mac cannot use process.stdin.fd (which isn't set up as sync)
+              var usingDevice = false;
+              try {
+                fd = fs.openSync('/dev/stdin', 'r');
+                usingDevice = true;
+              } catch (e) {}
+            }
 
-            bytesRead = fs.readSync(fd, buf, 0, BUFSIZE, null);
+            try {
+              bytesRead = fs.readSync(fd, buf, 0, BUFSIZE, null);
+            } catch(e) {
+              // Cross-platform differences: on Windows, reading EOF throws an exception, but on other OSes,
+              // reading EOF returns 0. Uniformize behavior by treating the EOF exception to return 0.
+              if (e.toString().indexOf('EOF') != -1) bytesRead = 0;
```

Comment:
I thought about that as well, but I suppose shouldn't be any harm to have in both, and prefer having as few extra chars of code here as possible. Node.js file system documentation at https://nodejs.org/api/fs.html doesn't seem to define EOF behavior at all (or I couldn't find any), so not sure which one would be the designed behavior. Posted this up at https://github.com/nodejs/node/issues/8024, folks there can probably chime in on that.